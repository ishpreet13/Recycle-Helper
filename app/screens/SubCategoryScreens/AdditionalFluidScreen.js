import React from "react";
import { View, StyleSheet, Text } from "react-native";

import AppButton from "../../components/AppButton";
import colors from "../../config/colors";

function AdditionalFluidScreen({ navigation }) {
  return (
    <View style={styles.container}>
      <Text style={styles.text}>Please choose the fluid category {"\n"} </Text>
      <View style={styles.buttonContainer}>
        <AppButton
          title="Auto Fluids"
          onPress={() =>
            navigation.navigate("InstructionScreen", {
              documentName: "Auto Fluids",
            })
          }
          fontSize={15}
        ></AppButton>
        <AppButton
          title="Cooking Oil"
          onPress={() =>
            navigation.navigate("InstructionScreen", {
              documentName: "Cooking Oil",
            })
          }
          fontSize={15}
        ></AppButton>
        <AppButton
          title="Used Motor Oil and Filters"
          onPress={() =>
            navigation.navigate("InstructionScreen", {
              documentName: "Used Motor Oil and Filters",
            })
          }
          fontSize={15}
        ></AppButton>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    backgroundColor: colors.white,
  },
  text: {
    color: colors.primary,
    fontSize: 22,
    fontWeight: "bold",
    textAlign: "center",
  },
  buttonContainer: {
    width: "90%",
    alignSelf: "center",
  },
});

export default AdditionalFluidScreen;
