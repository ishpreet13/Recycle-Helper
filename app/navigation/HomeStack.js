import React from "react";
import { createStackNavigator } from "@react-navigation/stack";

import HomeScreen from "../screens/HomeScreen";
import RecycleScreen from "../screens/RecycleScreen";
import UploadScreen from "../screens/UploadScreen";
import MainCategoryListScreen from "../screens/MainCategoryListScreen";
import PlasticScreen from "../screens/SubCategoryScreens/PlasticScreen";
import AutomotiveScreen from "../screens/SubCategoryScreens/AutomotiveScreen";
import ConstructionScreen from "../screens/SubCategoryScreens/ConstructionScreen";
import ElectronicsScreen from "../screens/SubCategoryScreens/ElectronicsScreen";
import GlassScreen from "../screens/SubCategoryScreens/GlassScreen";
import HouseholdWasteScreen from "../screens/SubCategoryScreens/HouseholdWasteScreen";
import HouseholdScreen from "../screens/SubCategoryScreens/HouseholdScreen";
import MetalScreen from "../screens/SubCategoryScreens/MetalScreen";
import PaperScreen from "../screens/SubCategoryScreens/PaperScreen";
import InstructionsScreen from "../screens/InstructionsScreen";
import PredictionScreen from "../screens/PredictionScreen";
import UpdateProfileScreen from "../screens/UpdateProfileScreen";
import UserProfileScreen from "../screens/UserProfileScreen";
import CongratulationsScreen from "../screens/CongratulationsScreen";
import ReportStack from "./ReportStack";
import checktrialsscreen from "../screens/checktrialsscreen";
import SetUserGoalScreen from "../screens/SetUserGoalScreen";
import UpdateGoalScreen from "../screens/UpdateGoalScreen";
import AdditionalFluidScreen from "../screens/SubCategoryScreens/AdditionalFluidScreen";

const Stack = createStackNavigator();

export default function HomeStack() {
  return (
    <Stack.Navigator screenOptions={{ headerShown: false }}>
      <Stack.Screen name="Home" component={HomeScreen} />
      <Stack.Screen name="Recycle" component={RecycleScreen} />
      <Stack.Screen name="Upload" component={UploadScreen} />
      <Stack.Screen
        name="MainCategoryList"
        component={MainCategoryListScreen}
      />
      <Stack.Screen name="Automotive" component={AutomotiveScreen} />
      <Stack.Screen name="Construction" component={ConstructionScreen} />
      <Stack.Screen name="Electronics" component={ElectronicsScreen} />
      <Stack.Screen name="Glass" component={GlassScreen} />
      <Stack.Screen name="Household" component={HouseholdScreen} />
      <Stack.Screen name="HouseholdWaste" component={HouseholdWasteScreen} />
      <Stack.Screen name="Metal" component={MetalScreen} />
      <Stack.Screen name="Paper" component={PaperScreen} />
      <Stack.Screen name="Plastic" component={PlasticScreen} />
      <Stack.Screen name="InstructionScreen" component={InstructionsScreen} />
      <Stack.Screen name="Prediction" component={PredictionScreen} />
      <Stack.Screen name="UpdateProfile" component={UpdateProfileScreen} />
      <Stack.Screen name="UserProfile" component={UserProfileScreen} />
      <Stack.Screen name="Congratulations" component={CongratulationsScreen} />
      <Stack.Screen name="Report" component={ReportStack} />
      <Stack.Screen name="trial" component={checktrialsscreen} />
      <Stack.Screen name="SetUserGoal" component={SetUserGoalScreen} />
      <Stack.Screen name="UpdateGoal" component={UpdateGoalScreen} />
      <Stack.Screen
        name="AdditionalFluidScreen"
        component={AdditionalFluidScreen}
      />
    </Stack.Navigator>
  );
}
