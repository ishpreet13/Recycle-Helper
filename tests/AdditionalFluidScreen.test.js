import "react-native";
import React from "react";
import renderer, { act } from "react-test-renderer";
import { render, fireEvent, waitFor } from "@testing-library/react-native";
import { NavigationContainer } from "@react-navigation/native";

import AdditionalFluidScreen from "../app/screens/SubCategoryScreens/AdditionalFluidScreen";
import HomeStack from "../app/navigation/HomeStack";

act(() => {
  describe("Testing AdditionalFluidScreen ", () => {
    let wrapper;
    it("should render", async () => {
      act(() => {
        wrapper = renderer.create(<AdditionalFluidScreen />);
      });
      expect(wrapper.toJSON()).toBeTruthy();
    });
  });
});

act(() => {
  it("AdditionalFluidScreen should render correctly", async () => {
    let tree;
    act(() => {
      tree = renderer.create(<AdditionalFluidScreen />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});

describe("Testing AdditionalFluidScreen  Style", () => {
  const wrapper = renderer.create(<AdditionalFluidScreen />);
  const styles = wrapper.toJSON().props.style;
  const flex = styles.flex;
  const backgroundColor = styles.backgroundColor;
  it("should have flex 1", () => {
    expect(flex).toBe(1);
  });
  it("should have backgroundColor #fff", () => {
    expect(backgroundColor).toBe("#fff");
  });
});

describe("AdditionalFluidScreen ", () => {
  it("navigates on button press", () => {
    const navigate = jest.fn();
    const { getByText } = render(
      <AdditionalFluidScreen navigation={{ navigate }} />
    );
    fireEvent.press(getByText("Auto Fluids"));
    expect(navigate).toHaveBeenCalledWith("InstructionScreen", {
      documentName: "Auto Fluids",
    });
  });
});

describe("AdditionalFluidScreen ", () => {
  it("navigates on button press", () => {
    const navigate = jest.fn();
    const { getByText } = render(
      <AdditionalFluidScreen navigation={{ navigate }} />
    );
    fireEvent.press(getByText("Used Motor Oil and Filters"));
    expect(navigate).toHaveBeenCalledWith("InstructionScreen", {
      documentName: "Used Motor Oil and Filters",
    });
  });
});

describe("AdditionalFluidScreen ", () => {
  it("navigates on button press", () => {
    const navigate = jest.fn();
    const { getByText } = render(
      <AdditionalFluidScreen navigation={{ navigate }} />
    );
    fireEvent.press(getByText("Cooking Oil"));
    expect(navigate).toHaveBeenCalledWith("InstructionScreen", {
      documentName: "Cooking Oil",
    });
  });
});
