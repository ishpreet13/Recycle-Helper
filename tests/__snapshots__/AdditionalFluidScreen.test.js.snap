// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdditionalFluidScreen should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "yellowgreen",
        "fontSize": 22,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    Please choose the fluid category 
    

     
  </Text>
  <View
    style={
      Object {
        "alignSelf": "center",
        "width": "90%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "yellowgreen",
          "borderColor": "#fff",
          "borderRadius": 25,
          "justifyContent": "center",
          "marginVertical": 10,
          "opacity": 1,
          "padding": 15,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 20,
              "fontWeight": "bold",
              "textAlign": "center",
              "textTransform": "uppercase",
            },
            Object {
              "fontSize": 15,
            },
          ]
        }
      >
        Auto Fluids
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "yellowgreen",
          "borderColor": "#fff",
          "borderRadius": 25,
          "justifyContent": "center",
          "marginVertical": 10,
          "opacity": 1,
          "padding": 15,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 20,
              "fontWeight": "bold",
              "textAlign": "center",
              "textTransform": "uppercase",
            },
            Object {
              "fontSize": 15,
            },
          ]
        }
      >
        Cooking Oil
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "yellowgreen",
          "borderColor": "#fff",
          "borderRadius": 25,
          "justifyContent": "center",
          "marginVertical": 10,
          "opacity": 1,
          "padding": 15,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 20,
              "fontWeight": "bold",
              "textAlign": "center",
              "textTransform": "uppercase",
            },
            Object {
              "fontSize": 15,
            },
          ]
        }
      >
        Used Motor Oil and Filters
      </Text>
    </View>
  </View>
</View>
`;
